include/boost/geometry/algorithms/detail/overlay/clip_linestring.hpp

```diff
@@ -51,14 +51,14 @@ class liang_barsky
 private:
     typedef model::referring_segment<Point> segment_type;
 
-    template <typename T>
-    inline bool check_edge(T const& p, T const& q, T& t1, T& t2) const
+    template <typename PointType, typename CalcType>
+    inline bool check_edge(PointType const& p, PointType const& q, CalcType& t1, CalcType& t2) const
```

Comment:
I suggest to change the name of PointType to CoordinateType since neither `p` nor `q` are not Points.